filter_plugins/linux_ufw_expand_rules.py

```diff
@@ -0,0 +1,237 @@
+import unittest
+from copy import deepcopy
+from typing import List, Dict
+
+
+def _linux_ufw_expand_rule_obj(obj: Dict, key: str, subkey: str, subkey_aliases: List) -> List[Dict]:
+    if not obj[key]:
+        return []
+    expanded_obj = []
+
+    for subkey_alias in subkey_aliases:
+        if subkey_alias in obj[key]:
+            if subkey in obj[key]:
+                if not isinstance(obj[key][subkey], list):
+                    obj[key][subkey] = [obj[key][subkey]]
+                if not isinstance(obj[key][subkey_alias], list):
+                    obj[key][subkey_alias] = [obj[key][subkey_alias]]
+                obj[key][subkey].extend(obj[key][subkey_alias])
+            else:
+                obj[key][subkey] = obj[key][subkey_alias]
+            del obj[key][subkey_alias]
+
+    if subkey in obj[key]:
+        if not isinstance(obj[key][subkey], list):
+            obj[key][subkey] = [obj[key][subkey]]
+        for value in obj[key][subkey]:
+            if value is None or value == '[AnsibleUndefined]':
+                continue
+            additional_obj = deepcopy(obj)
+            additional_obj[key][subkey] = value
+            expanded_obj.append(additional_obj)
+    else:
+        expanded_obj.append(obj)
+
+    return expanded_obj
+
+
+def _linux_ufw_expand_rules_key(rules: List, key: str, key_aliases=[]) -> List:
+    expanded_rules = []
+    for rule in rules:
+        expanded_rules.extend(_linux_ufw_expand_rule_obj(rule, 'value', key, key_aliases))
+    return expanded_rules
+
+
+def linux_ufw_expand_rules(rules: List) -> List:
+    rules = _linux_ufw_expand_rules_key(rules, 'from_ip', ['from'])
+    rules = _linux_ufw_expand_rules_key(rules, 'from_port')
+    rules = _linux_ufw_expand_rules_key(rules, 'to_ip', ['to'])
+    rules = _linux_ufw_expand_rules_key(rules, 'to_port', ['port'])
+    rules = _linux_ufw_expand_rules_key(rules, 'interface')
+    rules = _linux_ufw_expand_rules_key(rules, 'proto')
+    return rules
```

Comment:
I would put this part into its own class and make the "protected" functions into static private methods. In my opinion this is "cleaner code" and you don't have pseudo "protected" functions. This will also help with IDEs that have code support and code completion, because they will offer only the public parts of the class to calling objects. 
```suggestion
class LinuxUFW:
    
    @staticmethod
    def expand_rule_obj(obj: Dict, key: str, subkey: str, subkey_aliases: List) -> List[Dict]:
        if not obj[key]:
            return []
        expanded_obj = []

        for subkey_alias in subkey_aliases:
             if subkey_alias in obj[key]:
                if subkey in obj[key]:
                    if not isinstance(obj[key][subkey], list):
                        obj[key][subkey] = [obj[key][subkey]]
                    if not isinstance(obj[key][subkey_alias], list):
                        obj[key][subkey_alias] = [obj[key][subkey_alias]]
                    obj[key][subkey].extend(obj[key][subkey_alias])
                else:
                    obj[key][subkey] = obj[key][subkey_alias]
                del obj[key][subkey_alias]

        if subkey in obj[key]:
            if not isinstance(obj[key][subkey], list):
                obj[key][subkey] = [obj[key][subkey]]
            for value in obj[key][subkey]:
                if value is None or value == '[AnsibleUndefined]':
                    continue
                additional_obj = deepcopy(obj)
                additional_obj[key][subkey] = value
                expanded_obj.append(additional_obj)
        else:
             expanded_obj.append(obj)

        return expanded_obj

    @staticmethod
    def __expand_rules_key(rules: List, key: str, key_aliases=[]) -> List:
        expanded_rules = []
        for rule in rules:
            expanded_rules.extend(LinuxUFW.expand_rule_obj(rule, 'value', key, key_aliases))
        return expanded_rules

    @staticmethod
    def expand_rules(rules: List) -> List:
        rules = LinuxUFW.__expand_rules_key(rules, 'from_ip', ['from'])
        rules = LinuxUFW.__expand_rules_key(rules, 'from_port')
        rules = LinuxUFW.__expand_rules_key(rules, 'to_ip', ['to'])
        rules = LinuxUFW.__expand_rules_key(rules, 'to_port', ['port'])
        rules = LinuxUFW.__expand_rules_key(rules, 'interface')
        rules = LinuxUFW.__expand_rules_key(rules, 'proto')
        return rules
        
        
def linux_ufw_expand_rules(rules: List) -> List:
    return UFW.expand_rules(rules)
```